### ISSUE TITLE
Notice: Undefined offset: 1 On deserializing non-JWT string

### ISSUE THREAD

**Issue by scaytrase** [emarref/jwt#10]
https://github.com/emarref/jwt/blob/master/src/Serialization/Compact.php#L86  This code produses a notice when non-jwt string is passed for deserialization.

I think we could check this in a simple way

```php
$parts = explode('.', $jwt);
if (count($parts) !== 3) {
   throw new \InvalidArgumentException('Not a valid JWT string passed for deserialization');
}

list($encodedHeader, $encodedPayload, $encodedSignature) = $parts;
```

**Comment by tzetze** [comment]
It seems ok for me, the signature part could be emtpy, but JWT needs the second dot at the end in this case, so for me it seems checking the count against 3 is valid. May i create a merge request including tests for that?

**Comment by emarref** [comment]
Fixed by #14
